receiver/prometheusdiscoveryreceiver/internal/logger.go

```diff
@@ -0,0 +1,136 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal
+
+import (
+	gokitLog "github.com/go-kit/kit/log"
+	"github.com/go-kit/kit/log/level"
+	"go.uber.org/zap"
+)
+
+const (
+	levelKey = "level"
+	msgKey   = "msg"
+)
+
+// NewZapToGokitLogAdapter create an adapter for zap.Logger to gokitLog.Logger
+func NewZapToGokitLogAdapter(logger *zap.Logger) gokitLog.Logger {
```

Comment:
❤️ 

I take it other components might benefit from this! (Sidecar also wrangles go-kit logs a bunch.)

Reply:
I'll not take credits for this, it was copy-pasted from the prometheus receiver. As a side note we should at least make this shared between both receivers.